### PR TITLE
NodeJS_week2 HW uploaded

### PR DESCRIPTION
**Pull request by amjadiftikhar**

### CODE REVIEW

File: nodejs/week2/meal_sharing/src/backend/index.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+const route = express();
+const port = 3000;
+
+const mealsRoute = require("./routes/mealsRoute");
+const reservationsRoute = require("./routes/reservationsRoute");
+const reviewsRoute = require("./routes/reviewsRoute");
+
+// middleware function that logs out the time, a request was received and the path
+
+route.use((req, res, next) => {
```
Comment (benna100) [inline nodejs/week2/meal_sharing/src/backend/index.js]: Really nice!

File: nodejs/week2/meal_sharing/src/backend/routes/mealsRoute.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+route.get("/", (req, res) => {
+	res.json(meals);
+});
+
+//Respond with the json for the meal with the corresponding id
+
+route.get("/:id", (req, res) => {
+	const { id } = req.params;
+	const mealsById = meals.filter(meal => {
```
Comment (benna100) [inline nodejs/week2/meal_sharing/src/backend/routes/mealsRoute.js]: You can use array.find here

File: nodejs/week2/meal_sharing/src/backend/routes/mealsRoute.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+});
+
+// Get meals that has a price smaller than maxPrice
+// Get meals that partially match a title. Rød grød med will match the meal with the title Rød grød med fløde 
+// Get meals that has been created after the date   
+// Only specific number of meals
+
+
+route.get("/api/meals", (req, res) => {
+    const {maxPrice} = req.query
```
Comment (benna100) [inline nodejs/week2/meal_sharing/src/backend/routes/mealsRoute.js]: You can destructure multiple variables
const {maxPrice, title, ...} = req.query

File: nodejs/week2/meal_sharing/src/backend/index.js (hunk 1 above)
Comment (amjadiftikhar) [reply nodejs/week2/meal_sharing/src/backend/index.js]: Hi Benjamin 

Thank you for the feedback :) I will try to fix these due changes as soon as possible.

**Comment by amjadiftikhar** [comment]
all issues and comments have been fixed, thank you for the feedback

**Pull request merged**